app/services/config_report_importer.rb

```diff
@@ -0,0 +1,41 @@
+class ConfigReportImporter < ReportImporter
+  attr_reader :report
+  def self.authorized_smart_proxy_features
+    ['Puppet']
+  end
+
+  def report_name_class
+    ConfigReport
+  end
+
+  def import
+    @report = super
+    # Store all Puppet message logs
+    import_log_messages
+    # Check for errors
+    notify_on_report_error
+    logger.info("Imported report for #{name} in #{(Time.now - @start_time).round(2)} seconds")
+  end
+
+  private
+
+  def notify_on_report_error
```

Comment:
You might find this is a useful method to put in the base class, perhaps adding an argument to specify the notification name would make it reusable so other impls can seed a mail notification of their own and use it.